docs/contracts/v4/guides/04-hooks/swap.mdx

```diff
@@ -0,0 +1,235 @@
+---
+title: Swap
+---
+
+Swaps are the most common interaction with the Uniswap protocol. When it comes to swap there are two hook functions available to customize and extend its behavior:
+
+- `beforeSwap`
+- `afterSwap`
+
+As the names suggest `beforeSwap`/`afterSwap` are functions called before or after a swap is executed in a pool.
+
+This guide will explain the mechanism of encoded flags for hooks, and go through the parameters and examples for `beforeSwap` and `afterSwap`.
+
+Note: The swap examples are not production ready code, and are implemented in a simplistic manner for the purpose of learning.
+
+## Hook Flags
+
+As mentioned in [Concept of Hooks](../../concepts/04-hooks.mdx), hook contracts specify the permissions that determine which hook functions they implement by __encoding them in the address of the contract__. Then the `PoolManager` uses these permissions to determine which hook functions to call for a given pool based on its Key. 
+
+This is where encoded flags come into play. Each hook function e.g. `beforeSwap` - corresponds to a certain "flag". For example, the `beforeSwap` function is correlated to the [`BEFORE_SWAP_FLAG`](https://github.com/Uniswap/v4-core/blob/main/src/libraries/Hooks.sol#L37) which has a value of `1 << 7`.
+
+These flags represent specific bits in the address of the hook smart contract - and the value of the bit (a zero or a one) represents whether that flag is set to true or false. Let's understand this with an example:
+
+Addresses on Ethereum are 20 bytes long (160 bits). So for example the address:
+
+```
+0x00000000000000000000000000000000000000C0
+```
+
+represented in binary is:
+
+```solidity
+0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 
+0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 
+0000 0000 0000 0000 0000 0000 1100 0000
+```
+
+In binary it goes from right-to-left - so the trailing 8 bits of this address are `1100 0000` where:
+
+1st Bit to 6th Bit = `0`
+
+7th Bit to 8th Bit = `1`
+
+The `AFTER_SWAP` flag is represented by the 7th bit - which is set to `1` for the given contract address. That means this contract address - if it is a hook contract address - signals that the `afterSwap` function is implemented in it and the PoolManager will attempt to call that function during a `swap` workflow.
```

Comment:
```suggestion
The `AFTER_SWAP` flag is represented by the 7th bit - which is set to `1` for the example contract address. In the `PoolManager's` swap execution flow, it will observe the flag and make a call to the hook's `afterSwap` function.
```